ToolBox/SECDataDownloader/SECDataConverter.cs

```diff
@@ -123,189 +116,189 @@ public SECDataConverter(string rawSource, string destination, string knownEquity
 
-                            WriteReport(reports, ticker);
+                        // Encode all contents inside tags to prevent errors in XML parsing.
+                        // The json deserializer will convert these values back to their original form
+                        if (!parsingText && HasValue(newTextLine))
+                        {
+                            newTextLine =
+                                $"<{currentTagName}>{SecurityElement.Escape(GetTagValueFromLine(newTextLine))}</{currentTagName}>";
+                        }
+                        // Escape all contents inside TEXT tags
+                        else if (parsingText)
+                        {
+                            newTextLine = SecurityElement.Escape(newTextLine);
                         }
-                    );
 
-                    // This will clean up after ourselves without having to pay
-                    // the expense of deleting every single file inside the raw_data folder
-                    Directory.Delete(extractDataPath, true);
+                        // Don't encode the opening tag
+                        if (currentTagName == "TEXT")
+                        {
+                            parsingText = true;
+                        }
+
+                        xmlText.AppendLine(newTextLine);
+                    }
+
+                    ISECReport report;
+                    try
+                    {
+                        report = factory.CreateSECReport(xmlText.ToString());
+                    }
+                    // Ignore unsupported form types for now
+                    catch (DataException)
+                    {
+                        return;
+                    }
+                    catch (XmlException e)
+                    {
+                        Log.Error(e, $"SECDataConverter.Process(): Failed to parse XML from file path: {rawReportFilePath}");
+                        return;
+                    }
+                    catch (Exception e)
+                    {
+                        Log.Error(e, "SECDataConverter.Process(): Unknown error encountered");
+                        return;
+                    }
+
+                    // First filer listed in SEC report is usually the company listed on stock exchanges
+                    var companyCik = report.Report.Filers.First().CompanyData.Cik;
+
+                    // Some companies can operate under two tickers, but have the same CIK.
+                    // Don't bother continuing if we don't find any tickers for the given CIK
+                    List<string> tickers;
+                    if (!CikTicker.TryGetValue(companyCik, out tickers))
+                    {
+                        return;
+                    }
+
+                    try
+                    {
+                        // There can potentially not be an index file present for the given CIK
+                        GetPublicationDate(report, companyCik);
```

Comment:
Are there any other side effects required by invoking this function, or are we really just checking if:
``` csharp
var indexFilePath = Path.Combine(RawSource, "indexes", $"{companyCik}.json");
if (File.Exists(indexFilePath))
{
    throw new Exception($"Submission not found for CIK: {cik}");
}
```

If we're doing this just to check if the file exists, let's just check it  directly and _not_ throw the exception. Exceptions should be reserved for exceptional cases and [not for control flow](https://web.archive.org/web/20140430044213/http://c2.com/cgi-bin/wiki?DontUseExceptionsForFlowControl).